modules/local/kraken2/main.nf

```diff
@@ -0,0 +1,39 @@
+process KRAKEN2 {
+    tag "$meta.id"
+    label 'process_medium'
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' :
+        'biocontainers/mulled-v2-5799ab18b5fc681e75923b2450abaa969907ec98:87fc08d11968d081f3e8a37131c1f1f6715b6542-0' }"
+
+    input:
+    tuple val(meta), path(reads), path(kraken2_db)
+
+    output:
+    tuple val(meta), path("*_kraken2_output.tsv"), path("*_kraken2_report.txt"), emit:report
```

Comment:
Could you have each file emitted as a separate entry in `output`, with a separate name. Like:

```nextflow
tuple val(meta), path("*_kraken2_output.tsv"), emit: output_tsv
tuple val(meta), path("*_kraken2_report.txt"), emit: report_txt
```

That way these files can be accessed from channels separately using the names defined by `emit`.

Reply:
Change updated in [5bb1238](https://github.com/phac-nml/speciesabundance/pull/5/commits/5bb1238cfea7b8c220b1feedfac68fe05fb66eab)